CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/LayersViewController.java

```diff
@@ -38,142 +41,149 @@
         for (final LayerDescription layer : pane.getlayers()) {
-            if(!layer.getLayerQuery().equals(LayerDescription.DEFAULT_QUERY_STRING)){
+            if (!layer.getLayerQuery().equals(LayerDescription.DEFAULT_QUERY_STRING)) {
                 newBitmask |= layer.getCurrentLayerVisibility() ? (1 << layer.getLayerIndex() - 1) : 0;
             }
         }
         // if the newBitmask is 1, it means none of the boxes are checked. therefore display default layer 1 (All nodes)
         newBitmask = (newBitmask == 0) ? 0b1 : (newBitmask > 1) ? newBitmask & ~0b1 : newBitmask;
-        PluginExecution.withPlugin(new UpdateGraphBitmaskPlugin(newBitmask)).executeLater(GraphManager.getDefault().getActiveGraph());
+        PluginExecution.withPlugin(new UpdateGraphBitmaskPlugin(newBitmask))
+                .executeLater(GraphManager.getDefault().getActiveGraph());
     }
 
     /**
-     * Grab all queries entered into text areas and store them in the qraph's
-     * queries.
+     * Get all layer queries from the Layer View and store them on the qraph.
      */
     public void submit() {
-        // ensure pane is set to the content of the parent view.
-        pane = parent.getContent();
+        final LayersViewPane pane = parent.getContent();
         if (pane == null) {
             return;
         }
         final List<String> layerQueries = new ArrayList<>();
-        for (LayerDescription layer : pane.getlayers()) {
-            if(!layer.getLayerQuery().equals(LayerDescription.DEFAULT_QUERY_STRING)){
-                layerQueries.add(layer.getLayerQuery() == "" ? null : layer.getLayerQuery());
+        for (final LayerDescription layer : pane.getlayers()) {
+            if (!layer.getLayerQuery().equals(LayerDescription.DEFAULT_QUERY_STRING)) {
+                layerQueries.add(layer.getLayerQuery().isEmpty() ? null : layer.getLayerQuery());
             }
         }
-        PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(layerQueries)).executeLater(GraphManager.getDefault().getActiveGraph());
+        PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(layerQueries))
+                .executeLater(GraphManager.getDefault().getActiveGraph());
     }
 
     /**
-     * Executes the plugin with parameters dependent on if the trigger was an update event.
+     * Reads the graph's layers_view_state attribute and populates the Layers
+     * View pane.
      */
-    public void updateState(final List<LayerDescription> layers, final boolean wasUpdate) {
-        pane = parent.getContent();
-        stateChanged = true;
-        PluginExecution.withPlugin(new LayersViewStateUpdater(pane, layers, wasUpdate)).executeLater(GraphManager.getDefault().getActiveGraph());
+    public void readState() {
+        final LayersViewPane pane = parent.getContent();
+        if (pane == null) {
+            return;
+        }
         
+        final Graph graph = GraphManager.getDefault().getActiveGraph();
+        if(graph == null){
```

Comment:
This if statement and the one above can be merged into one: `if (pane == null || graph == null)`